docs/proposals/central-segmented-registry.md

```diff
@@ -0,0 +1,144 @@
+
+### Managing Images
+
+ImageStreamTags are annotated (annotations or labels?) with a list of approved registry environments. Users may promote (or demote) images by managing the environment whitelist. For backward compatibility, the default environment is "default". For example:
+
+        kind: ImageStreamTag
+          annotations:
+            environment:
+              - test
+              - uat
+
+To promote to the production environment, a user or process adds an annotation to the ImageStreamTag object. This action would trigger a job to manage ImageStream metadata, registry and storage so the image could be served from the desired registry environment.
+
+        kind: ImageStreamTag
+          annotations:
+            environment:
+              - test
+              - uat
+              - prod
+
+### Managing the Registries
+
+The central registry cluster may have one or many registries deployed. Each registry is deployed into a project matching the environment name (e.g. test, uat, prod). For example:
+
+        $ oc new-project test
+        $ oadm registry --namespace test
+        $ oc expose service docker-registry --namespace test
+
+The exposed route provides a unique endpoint for referencing the registry.
+
+Each registry has its own storage backend.
+
+### Managing the Clusters
+
+The central registry is referenced using external service abstraction. Instead of deploying a registry on each cluster, Service and Route objects are created in the "default" project namespace on each cluster.
+
+        - kind: Service
+          apiVersion v1
+          metadata:
+            name: docker-registry
+          spec:
+            ports:
+              - name: docker-registry
+                protocol: TCP
+                port: 443
+                targetPort: 443
+                nodePort: 0
+          selector: {}
+        - kind: Route
+          apiVersion: v1
+          metadata:
+            labels:
+              app: docker-registry
+            name: docker-registry
+          spec:
+            tls:
+              ...
+            to:
+              kind: Service
+              name: docker-registry
+          status: {}
+
+An Endpoint object created in the "default" project namespace on each cluster. In this example, the test OpenShift cluster references the test registry instance in the central registry cluster.
+
+        kind: Endpoints
+        apiVersion: v1
+        metadata:
+          name: docker-registry
+        subsets:
+          - addresses:
+              - IP: docker-registry-test.apps.example.com
+            ports:
+              - port: 443
+                name: docker-registry
+
+### Managing Authorization
+
+Cross-cluster authorization is a significant challenge. Automating or synchronizing cross-cluster authorization is ultimately desired. Until that can be achieved, manual processes may need to bridge the gap.
```

Comment:
In practice, most "production" clusters have fairly homogenous auth (more privileged).  If production can pull a wider set of content, that's usually ok.